### DISCUSSION TITLE
Increase the clock speed?

### DISCUSSION THREAD

**Discussion by VerdichteteEnergie** [hneemann/Digital#1271] [General]
Is there a way to increase the clock speed so that the output is displayed immediately?

My problem: I rebuilt the computer from the nand2tetris course. An operating system is required for the programs. Because the operating system has to load, it takes a long time to display the output, even with the maximum value of clock speed.

**Comment by hneemann** [comment]
No! If the clock rate is set to the maximum value, this means that one of the cores is 100% occupied with the simulation. The visualization is performed on a second core without slowing down the first. I have not yet found a way to efficiently distribute the actual simulation over several cores.
Is it not possible to simply store the operating system in a ROM? Then it would be available straight away.

**Reply by VerdichteteEnergie** [reply]
Thanks for the explanation! My CPU is a bit older (FX-6300), but the clock rate of one core should still be enough for 16-bit programs, right? Or does the simulation also cost computing power (sorry for the noob question)?

I use ROM, but I see that it's probably not the clock rate, but rather I was caught in a loop. Apparently I need to further debug my code for the operating system. So far I had only used the nand2tetris emulator for this. I finished the course a long time ago. In the end, my goal was always to be able to create the Hack-Computer from the course myself using your excellent program. It's a project that I do on the side and I really enjoy.

Offtopic: Thank you for your work! It's really great that you maintain your program so much and even often answer the comments yourself.
